mlflow/store/runs_artifact_repo.py

```diff
@@ -0,0 +1,96 @@
+from six.moves import urllib
+
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact_repo import ArtifactRepository
+
+
+class RunsArtifactRepository(ArtifactRepository):
+    """
+    Handles artifacts associated with a Run via URIs of the form
+      `runs:/<run_id>/run-relative/path/to/artifact`.
+    It is a light wrapper that resolves the artifact path to an absolute URI then instantiates
+    and uses the artifact repository for that URI.
+
+    The relative path part of ``artifact_uri`` is expected to be in posixpath format, so Windows
+    users should take special care when constructing the URI.
+    """
+
+    def __init__(self, artifact_uri):
+        from mlflow.tracking.artifact_utils import get_artifact_uri
+        from mlflow.store.artifact_repository_registry import get_artifact_repository
+        (run_id, artifact_path) = RunsArtifactRepository.parse_runs_uri(artifact_uri)
+        uri = get_artifact_uri(run_id, artifact_path)
+        assert urllib.parse.urlparse(uri).scheme != "runs"  # avoid an infinite loop
+        super(RunsArtifactRepository, self).__init__(artifact_uri)
+        self.repo = get_artifact_repository(uri)
+
+    @staticmethod
+    def parse_runs_uri(run_uri):
+        parsed = urllib.parse.urlparse(run_uri)
+        if parsed.scheme != "runs":
+            raise MlflowException("Not a proper runs:/ URI: %s" % run_uri)
```

Comment:
Nit: maybe we could include the format of a valid runs URI in the error message, like:

`"Not a proper runs:/ URI: %s. Runs URIs must be of the form 'runs:/<run_id>/run-relative/path/to/artifact'" % run_uri`